internet-programming/hw2/server/Server.java

```diff
@@ -0,0 +1,95 @@
+package hw2.server;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousChannelGroup;
+import java.nio.channels.AsynchronousServerSocketChannel;
+import java.nio.channels.AsynchronousSocketChannel;
+import java.nio.channels.CompletionHandler;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class Server {
+
+    private static final int PORT = 31337;
+    private static final int BUFFER_SIZE = 1024;
+
+    public static void main(String[] args) throws IOException {
+        ExecutorService threadPool = Executors.newCachedThreadPool();

```

Comment:
As we already speak about that Executors.newCachedThreadPool() with an AsynchronousChannelGroup may lead to high resource usage as the thread pool grows dynamically. A fixed thread pool size would be preferable to limit the resource footprint.